forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt

```diff
@@ -28,28 +28,28 @@ internal object LDContextKind {
 internal object LDManager {
     private const val LD_MOBILE_KEY = BuildConfig.LD_MOBILE_KEY
     private var internalVaultType: String? = null
-    private var logger = Logger.getInstance(BuildConfig.DEBUG)
+    private var internalLogger: Log? = null
 
     internal var vaultType: String?
         get() = internalVaultType
         set(value) {
             if (value == null) {
-                logger.warning("vaultType is being reset to null. This should only happen while unit testing!")
+                internalLogger?.w("vaultType is being reset to null. This should only happen while unit testing!")
                 internalVaultType = null
             } else if (internalVaultType == null) {
                 internalVaultType = value
-            } else {
-                logger.warning("vaultType can only be set once!")
             }
+            internalLogger?.e("vaultType can only be set once!")
```

Comment:
This setter was exposed entirely for testing purposes. I agree that throwing in the scenario that someone is trying to change a non-null value to another non-null value would make sense and adding comments would be beneficial.

Also, the `LDManager` is following the singleton pattern by not exposing any form of constructor and managing a static version of itself. We can't accept an `internalVaultType` because that value isn't set until the `LDManager` is instantiated and gets the value.